src/lib/utils/getSarcophagusState.ts

```diff
@@ -0,0 +1,29 @@
+import { ethers } from 'ethers';
+import { SarcophagusResponseContract, SarcophagusState } from 'types';
+
+export const getSarcophagusState = (
+  sarco: SarcophagusResponseContract,
+  gracePeriod: number
+): SarcophagusState => {
+  if (sarco.resurrectionTime.eq(ethers.constants.Zero)) return SarcophagusState.DoesNotExist;
+  if (sarco.resurrectionTime.eq(ethers.constants.MaxUint256)) return SarcophagusState.Buried;
+  if (sarco.isCompromised) return SarcophagusState.Accused;
+
+  const nowSeconds = Math.trunc(Date.now() / 1000);
+  const withinGracePeriod =
+    nowSeconds >= sarco.resurrectionTime.toNumber() &&
+    nowSeconds < sarco.resurrectionTime.toNumber() + gracePeriod;
+
+  if (withinGracePeriod) return SarcophagusState.Resurrecting;
+
+  const isPastGracePeriod = nowSeconds >= sarco.resurrectionTime.toNumber() + gracePeriod;
+  const isCleaned = isPastGracePeriod && !sarco.hasLockedBond;
+
+  if (sarco.publishedKeyShareCount >= sarco.threshold)
```

Comment:
@DarksightKellar -- this should be above line 12 --- if we know there's enough keyshares published, then we can return `Resurrected`. Otherwise, we can check if we are still resurrecting.